src/lib/operators/index_scan.cpp

```diff
@@ -38,28 +36,56 @@ std::shared_ptr<const Table> IndexScan::_on_execute() {
 
   _out_table = std::make_shared<Table>(_in_table->column_definitions(), TableType::References);
 
-  std::mutex output_mutex;
+  PartialHashIndex::Iterator range_begin;
+  // auto range_end = AbstractChunkIndex::Iterator{};
 
-  auto jobs = std::vector<std::shared_ptr<AbstractTask>>{};
-  if (included_chunk_ids.empty()) {
-    const auto chunk_count = _in_table->chunk_count();
-    jobs.reserve(chunk_count);
-    for (auto chunk_id = ChunkID{0u}; chunk_id < chunk_count; ++chunk_id) {
-      const auto chunk = _in_table->get_chunk(chunk_id);
-      Assert(chunk, "Physically deleted chunk should not reach this point, see get_chunk / #1686.");
+  auto matches_out = std::make_shared<RowIDPosList>();
 
-      jobs.push_back(_create_job(chunk_id, output_mutex));
+  const auto& indexes = _in_table->get_table_indexes();
+  Assert(!indexes.empty(), "No indexes for the requested ColumnID available.");
+
+  if (indexes.size() > 1) {
+    PerformanceWarning("There are multiple indexes available, but only the first one is used.");
+  }
```

Comment:
Due to #2348, I would suggest not introducing new PerformanceWarnings anymore.

Reply:
Good point.
I changed it to an Assert.